spec/services/group_stage_service_spec.rb

```diff
@@ -72,4 +77,72 @@
       end
     end
   end
+
+  describe '#update_group_scores' do
+    shared_examples_for 'only_return_group_scores' do
+      it 'only returns group_scores' do
+        @changed_group_scores.each do |gs|
+          expect(gs).to be_a(GroupScore)
+        end
+      end
+    end
+
+    context 'with only undecided matches' do
+      before do
+        @group = create(:group, match_factory: :running_group_match)
+        @group.matches.each do |match|
+          match.match_scores.each do |ms|
+            ms.points = 42
+            ms.save!
+          end
+        end
+        @changed_group_scores = GroupStageService.update_group_scores(@group)
+      end
+
+      it 'assigns 1 point to every team' do
+        @changed_group_scores.map(&:group_points).each do |points|
+          expect(points).to be(1)
+        end
+      end
+
+      it_should_behave_like 'only_return_group_scores'
```

Comment:
imo this should be a the top (or end) of this context to avoid confusion

Reply:
done